source/val/BasicBlock.h

```diff
@@ -61,22 +75,40 @@ class BasicBlock {
   /// Returns the successors of the BasicBlock
   std::vector<BasicBlock*>* get_successors() { return &successors_; }
 
-  /// Returns true if the  block should be reachable in the CFG
+  /// Returns true if the block is reachable in the CFG
   bool is_reachable() const { return reachable_; }
 
+  /// Returns the type of the BasicBlock
+  bool is_type(BlockType type) const { return type_.test(type); }
+
+  /// Sets the reacability of the basic block in the CFG
   void set_reachability(bool reachability) { reachable_ = reachability; }
 
+  /// Sets the type of the BasicBlock
+  void set_type(BlockType type) { type_.set(static_cast<int>(type)); }
```

Comment:
Minor.  BlockType has underlying type uint32_t.   It would be slightly better to use uint32_t here instead of static cast to int.
And it's probably too baroque to use std::underlying_type.  Yes, too baroque.

(still reviewing...)